benchmark/atoi.h

```diff
@@ -19,21 +19,15 @@ std::size_t c_strLength(const char *s);
 std::size_t c_strLength_natural(const char *s);
 int32_t c_strToI(const char *) noexcept;
 int64_t c_strToL(const char *) noexcept;
+int64_t c_strToL128(const char *) noexcept;
 
-inline int compareAtoi(const char *s) {
-    auto
-        from_stdlib = atoi(s),
-        from_zoo = c_strToI(s);
-    if(from_stdlib != from_zoo) { throw 0; }
-    return from_stdlib;
-}
-
+template<long long (*FUN)(const char *s)>
```

Comment:
i love this. 

why not extend the abstaction? 

```cpp
template<long long (*FUN_A)(const char *s), long long (*FUN_B)(const char *s)>
inline int compareStringConsumer(const char *s) {
    auto
        a = FUN_A(s),
        b = FUN_B(s);
    if(a != b) {
        auto a = FUN_A(s);
        throw 0; }
    return from_stdlib;
}
```

Reply:
i'm sure you could generalise it even more than that to be fair. you could make this compare any single arg functions where the return type has operator== implemented. 

```cpp
auto result = compareFunctions<strlen, zoo_strlen>(input);
```

Reply:
Yes, I will generalize as requested, leave this conversation "unresolved"